docs/integrations/sources/s3.md

```diff
@@ -152,66 +172,105 @@ Please note, the S3 Source connector used to infer schemas from all the availabl
+- **Quote Character**: In some cases, data values may contain instances of reserved characters \(like a comma, if that's the delimiter\). CSVs can handle this by wrapping a value in defined quote characters so that on read it can parse it correctly. By default, this is set to `"`.
+- **Escape Character**: An escape character can be used to prefix a reserved character and ensure correct parsing. Leaving this field blank (default option) will disallow escaping.
+- **Encoding**: Some data may use a different character set \(typically when different alphabets are involved\). See the [list of allowable encodings here](https://docs.python.org/3/library/codecs.html#standard-encodings). By default, this is set to `utf8`.
+- **Double Quote**: This option determines whether two quotes in a quoted CSV value denote a single quote in the data. Set to True by default.
+- **Allow newlines in values**: Also known as _multiline_, this option addresses situations where newline characters occur within text data. Typically, newline characters signify the end of a row in a CSV, but when this option is set to True, parsing correctly handles newlines within values. Set to False by default.
+- **Additional Reader Options**: This allows for editing the less commonly used CSV [ConvertOptions](https://arrow.apache.org/docs/python/generated/pyarrow.csv.ConvertOptions.html#pyarrow.csv.ConvertOptions). The value must be a valid JSON string, e.g.:
+
+  ```
+  {
+    "timestamp_parsers": [
+    "%m/%d/%Y %H:%M", "%Y/%m/%d %H:%M"
+    ], 
+    "strings_can_be_null": true, 
+    "null_values": [
+      "NA", "NULL"
+    ]
+  }
   ```
 
-- `advanced_options` : This allows for editing the less commonly required CSV [ReadOptions](https://arrow.apache.org/docs/python/generated/pyarrow.csv.ReadOptions.html#pyarrow.csv.ReadOptions). The value must be a valid JSON string. One use case for this is when your CSV has no header, or you want to use custom column names, you can specify `column_names` using this option.
+- **Advanced Options**: This allows for editing the less commonly used CSV [ReadOptions](https://arrow.apache.org/docs/python/generated/pyarrow.csv.ReadOptions.html#pyarrow.csv.ReadOptions). The value must be a valid JSON string. One use case for this is when your CSV has no header, or if you want to use custom column names. You can specify `column_names` using this option. For example:
 
-  ```test
-  {"column_names": ["column1", "column2", "column3"]}
   ```
+  {
+    "column_names": [
+      "column1", "column2", "column3"
+    ]
+  }
+  ```
+
+- **Block Size**: This is the number of bytes to process in memory at a time while reading files. The default value of `10000` is usually suitable, but if your table is particularly wide \(lots of columns / data in fields is large\), increasing this might help avoid schema detection failures.
```

Comment:
```suggestion
- **Block Size**: This is the number of bytes to process in memory at a time while reading files. The default value of `10000` is usually suitable, but if your files are particularly wide \(lots of columns or the values in the columns are particularly large), increasing this might help avoid schema detection failures.
```